internal/provisioner/teleport.go

```diff
@@ -88,15 +91,35 @@ func (n *teleport) ActualVersion() string {
 }
 
 func (n *teleport) Destroy() error {
+	teleportClusterName := fmt.Sprintf("cloud-%s-%s", n.environment, n.cluster.ID)
+	err := n.awsClient.S3EnsureBucketDeleted(teleportClusterName, n.logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to delete Teleport bucket")
+	}
+
+	err = n.awsClient.DynamoDBEnsureTableDeleted(teleportClusterName, n.logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to delete Teleport dynamodb table")
+	}
+
+	err = n.awsClient.DynamoDBEnsureTableDeleted(fmt.Sprintf("%s-events", teleportClusterName), n.logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to delete Teleport dynamodb events table")
+	}
 	return nil
 }
 
 func (n *teleport) NewHelmDeployment() *helmDeployment {
+	awsRegion := os.Getenv("AWS_REGION")
+	if awsRegion == "" {
+		awsRegion = aws.DefaultAWSRegion
+	}
```

Comment:
I am pretty sure that the awsClient passed in to `teleport` has the region set already. We should be able to just use that. https://github.com/mattermost/mattermost-cloud/blob/3006530cca13f09dc8552dec515710221ebfc79a/cmd/cloud/server.go#L165

Reply:
@gabrieljackson that's what I thought as well but I couldn't find any reference. Do you have any example?

Reply:
After discussing this offline with @stylianosrigas, we will keep this as is and will do a follow-up to clean this up a bit at a later date.